imap_processing/ena_maps/ena_maps.py

```diff
@@ -867,6 +870,304 @@ def __init__(
+
+        Takes a rectangular pixel, and recursively breaks it up into
+        smaller and smaller subpixels, then calculates the solid-angle weighted mean
+        of the healpix map's value at this pixel, until the difference
+        between the mean values of two consecutive subdivisions is within the
+        specified tolerances. The function returns the mean value at the final level
+        of subdivision and the depth of recursion.
+
+        Parameters
+        ----------
+        rect_pix_center_lon_lat : np.typing.NDArray | tuple[float, float]
+            The center longitude and latitude of the rectangular pixel.
+        rect_pix_spacing_deg : float
+            The spacing of the rectangular pixel in degrees.
+        value_key : str
+            The name of the value to interpolate from the healpix map.
+        tolerances : tuple[float, float], optional
+            The relative and absolute tolerances for convergence,
+            by default (1e-3, 1e-12).
+        max_subdivision_depth : int, optional
+            The maximum depth of recursion for subdivision,
+            by default MAX_SUBDIV_RECURSION_DEPTH.
+            Computation grows exponentially with depth, but only where the value
+            has a significant gradient between adjacent healpix pixels.
+            If the value is smooth, the recursion depth will be low.
+
+        Returns
+        -------
+        tuple[list[float], int]
+            The mean value at the final level of subdivision and the depth of recursion.
+        """
+        relative_tolerance, absolute_tolerance = tolerances
+
+        # Recursively subdivide a pixel and calculate its mean value until either the
+        # difference between consecutive levels is within the specified tolerances
+        # or the maximum recursion depth is reached
+        depth = 0
+        mean_pixel_value_at_level = []
+        while depth < max_subdivision_depth:
+            mean_pixel_value = (
+                self.calculate_rect_pixel_value_from_healpix_map_n_subdivisions(
+                    rect_pix_center_lon_lat=rect_pix_center_lon_lat,
+                    rect_pix_spacing_deg=rect_pix_spacing_deg,
+                    value_key=value_key,
+                    num_subdivisions=depth,
+                )
+            )
+            mean_pixel_value_at_level.append(mean_pixel_value)
+
+            # Determine if tolerance is met
+            # (skip on the 0th iteration, as there's no delta)
+            if depth > 0:
+                abs_delta = np.abs(
+                    mean_pixel_value_at_level[-1].mean()
+                    - mean_pixel_value_at_level[-2].mean()
+                )
+                total_abs_tolerance = (
+                    relative_tolerance * np.abs(mean_pixel_value_at_level[-1].mean())
+                    + absolute_tolerance
+                )
+                if abs_delta < total_abs_tolerance:
```

Comment:
Can you use numpy's `isclose()` here to use their relative / absolute tolerances?
https://numpy.org/doc/stable/reference/generated/numpy.isclose.html